src/mono/mono/metadata/class.c

```diff
@@ -2001,7 +2001,7 @@ mono_method_get_vtable_slot (MonoMethod *method)
 			g_assert (klass_methods);
 			mcount = mono_class_get_method_count (method->klass);
 			for (i = 0; i < mcount; ++i) {
-				if (klass_methods [i] == method)
+				if (klass_methods [i] == method || (klass_methods [i]->name == method->name && klass_methods [i]->slot != -1))
 					break;
```

Comment:
Why is this needed ?

Reply:
In a fullaot runtime test with generic methods, a vtable slot can't be found by method's address but by name

Reply:
This shouldn't be needed, every method is stored in klass->methods of some class, so either the class we are searching is wrong, or its a generic method instance, in which case we should search for its definition.

Reply:
It is a generic method instance and we search for its definition, but when we compare `klass_methods [i] == method` they don't match. They have the same name, so it should be the same instances.  What could result in different instances of the same method?

Reply:
Discovered that instance method is passed, but since its class is a generic interface, `method->declaring` is skipped and its slot can't be found. It should be fine if we pass `((MonoMethodInflated*)method)->declaring`